### PR TITLE
Support imports

### PR DESCRIPTION
**Pull request by djhaskin987-at-sling**
The purpose of this PR is to add support for importing into terraform. It's in it's early stages, but it feels like it's working so far. What do you think?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,19 +25,45 @@ func main() {
 	})
 }
 
+func normalizeInput(input string) (string, error) {
```
Comment (djhaskin987-at-sling) [inline main.go]: This function makes sure that the content can be updated by making sure yaml from kubernetes and hand-edited YAML is compared by terraform as the same.

File: main.go [hunk 2]
```diff
@@ -158,8 +186,43 @@ func resourceManifestRead(d *schema.ResourceData, m interface{}) error {
 	if err := run(cmd); err != nil {
 		return err
 	}
-	if strings.TrimSpace(stdout.String()) == "" {
+	output := strings.TrimSpace(stdout.String())
+	if output == "" {
 		d.SetId("")
+		return nil
+	}
```
Comment (djhaskin987-at-sling) [inline main.go:204]: The idea here is I grab the contents of the annotation `kubectl.kubernetes.io/last-applied-configuration`, normalize it, and read the content in using that. It should be correct in how it works. This allows for both terraform refresh and terraform import.

**Comment by ericchiang** [comment]
Out of curiosity, why? From the docs:

> prior to running terraform import it is necessary to write manually a resource configuration block for the resource, to which the imported object will be attached.

If you already have to define your resources, why not just define them and `apply` them?

**Comment by djhaskin987-at-sling** [comment]
Two reasons:

1) If terraform state is ever erroneously deleted, I want to be able to re-import what is already deployed to kubernetes without deleting it first.
2) If I have been using kubectl without terraform heretofore, but then I decide I want to use terraform, I want to be able to have a migration path for putting stuff into terraform *that has already been deployed* without having to delete the deployment first simply so that I can re-deploy it with terraform.

The PR also allows for ID-only refreshes, which is nice. :)

**Comment by ericchiang** [comment]
I really appreciate the development time, but I haven't agreed that this is a feature I want as part of this plugin :)

To save you from more coding without a promise to merge, I'm going to close this PR.

Can you please open an issue explaining:
* What you want to support and why
* What the use cases for this feature is
  * A quick example of what using that feature looks like
* The hand-wavy technical approach to implement this

**Pull request closed without merging**